pymeasure/instruments/tdk/tdk_base.py

```diff
@@ -257,29 +243,34 @@ def write(self, command):
 
         Valid frequency values are 18, 23, or 46 Hz. Default value is 18 Hz.
         """,
+        check_set_errors=True,
         validator=strict_discrete_set,
         values=[18, 23, 46]
     )
 
-    source_output = Instrument.control(
+    output_enabled = Instrument.control(
         "OUT?", "OUT %s",
         """Control the output of the power supply.
 
-        Valid values are ``'ON'`` and ``'OFF'``.
+        Valid values are ``True`` and ``False``.
         """,
+        check_set_errors=True,
         validator=strict_discrete_set,
-        values=["ON", "OFF"]
+        values={True: "ON", False: "OFF"},
+        map_values=True
     )
 
     foldback = Instrument.control(
```

Comment:
if you rename it to
```suggestion
    foldback_enabled = Instrument.control(
```
the values True/False are self evident.

Reply:
Updated in last commit